src/main/java/frc/robot/subsystems/hood/Hood.java

```diff
@@ -0,0 +1,98 @@
+package frc.robot.subsystems.hood;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.Logger;
+import java.util.function.Supplier;
+
+import static edu.wpi.first.units.Units.Radians;
+
+public class Hood extends SubsystemBase {
+    private static Hood INSTANCE = null;
+    private final HoodInputsAutoLogged inputs = HoodIO.inputs;
+    private final HoodIO io;
+    private final Mechanism2d mechanism2d = new Mechanism2d(
+            3,3
+    );
+    private final MechanismRoot2d root = mechanism2d.getRoot("Hood", HoodConstants.mechanism2dPose.getFirst(), HoodConstants.mechanism2dPose.getSecond());
+    private final MechanismLigament2d hood = root.append(
+        new MechanismLigament2d("Hood", HoodConstants.HoodLength, 45)
+    );
+
+    private ControlMode controlMode = ControlMode.POSITION;
+
+    /**
+     * Constructor for Hood subsystem.
+     *
+     * @param io IO of the subsystem.
+     */
+    private Hood(HoodIO io) {
+        this.io = io;
+    }
+
+    /**
+     * Gets the single instance of hoodSubsystem.
+     *
+     * @return The single instance of hoodSubsystem.
+     */
+    public static Hood getInstance() {
+        return INSTANCE;
+    }
+
+    public static void initialize(HoodIO io) {
+        INSTANCE = new Hood(io);
+    }
+
+    public MutableMeasure<Angle> getAngle() {
+        return inputs.angle;
+    }
+
+    public boolean atSetpoint() {
+        return io.atSetpoint();
+    }
+
+    public Command setAngle(Supplier<Rotation2d> angle){
```

Comment:
There's no reason to receive `Rotation2d` but internally use `Measure` objects. Either use `Rotation2d` everywhere or `Measure` everywhere.

Reply:
ae4a442